prereise/gather/griddata/transmission/geometry.py

```diff
@@ -0,0 +1,286 @@
+
+        if self.resistance_per_km is None:
+            try:
+                self.resistivity = resistivity[self.material]
+            except KeyError:
+                raise ValueError(
+                    f"Unknown resistivity for {self.material}, "
+                    "can't calculate resistance"
+                )
+            if self.area is None:
+                self.area = pi * self.radius**2
+            # convert per-m to per-km
+            self.resistance_per_km = self.resistivity * 1000 / self.area
+
+
+@dataclass
+class ConductorBundle(DataclassWithValidation):
+    """Represent a bundle of conductors (or a 'bundle' of one).
+
+    :param int n: number of conductors in bundle (can be one).
+    :param Conductor conductor: information for each conductor.
+    :param float spacing: distance between the centers of each conductor (meters).
+    :param str layout: either 'circular' (conductors are arranged in a regular polygon
+        with edge length ``spacing``) or 'flat' (conductors are arranged in a line, at
+        regular spacing ``spacing``).
+    """
+
+    n: int
+    conductor: Conductor
+    spacing: float = None  # we need to be able to ignore spacing for a single conductor
+    layout: str = "circular"
+    resistance_per_km: float = field(init=False)
+    spacing_L: float = field(init=False)  # noqa: N815
+    spacing_C: float = field(init=False)  # noqa: N815
+
+    def __post_init__(self):
+        self.validate_input_types()  # defined in DataclassWithValidation
+        self.resistance_per_km = self.conductor.resistance_per_km / self.n
+        self.spacing_L = self.calculate_equivalent_spacing("inductance")
+        self.spacing_C = self.calculate_equivalent_spacing("capacitance")
+
+    def calculate_equivalent_spacing(self, type="inductance"):
+        if type == "inductance":
+            conductor_distance = self.conductor.gmr
+        elif type == "capacitance":
+            conductor_distance = self.conductor.radius
+        else:
+            raise ValueError("type must be either 'inductance' or 'capacitance'")
+        if self.n == 1:
+            return conductor_distance
+        elif self.n == 2:
+            return (conductor_distance * self.spacing) ** (1 / 2)
+        else:
+            if self.layout == "circular":
+                return self.calculate_equivalent_spacing_circular(conductor_distance)
+            if self.layout == "flat":
+                return self.calculate_equivalent_spacing_flat(conductor_distance)
+            raise ValueError(f"Unknown layout: {self.layout}")
+
+    def calculate_equivalent_spacing_circular(self, conductor_distance):
+        if self.n == 3:
+            return (conductor_distance * self.spacing**2) ** (1 / 3)
+        if self.n == 4:
+            return (conductor_distance * self.spacing**3 * 2 ** (1 / 2)) ** (1 / 4)
+        raise NotImplementedError("Geometry calculations not implemented for n > 4")
```

Comment:
Maybe we can say `Geometry calculations are implemented for 1 <= n <= 4` to cover corner cases for non-positive integers.

Reply:
Done.